cli/src/main/java/com/devonfw/tools/ide/tool/npm/Npm.java

```diff
@@ -0,0 +1,60 @@
+package com.devonfw.tools.ide.tool.npm;
+
+import java.nio.file.Path;
+import java.util.Set;
+
+import com.devonfw.tools.ide.common.Tag;
+import com.devonfw.tools.ide.context.IdeContext;
+import com.devonfw.tools.ide.io.FileAccess;
+import com.devonfw.tools.ide.tool.LocalToolCommandlet;
+import com.devonfw.tools.ide.tool.ToolCommandlet;
+import com.devonfw.tools.ide.tool.node.Node;
+
+/**
+ * {@link ToolCommandlet} for <a href="https://nodejs.org/">node</a>.
+ */
+
+public class Npm extends LocalToolCommandlet {
+
+  /**
+   * The constructor.
+   *
+   * @param context the {@link IdeContext}.
+   */
+  public Npm(IdeContext context) {
+
+    super(context, "npm", Set.of(Tag.JAVA_SCRIPT, Tag.RUNTIME));
+  }
+
+  @Override
+  public boolean install(boolean silent) {
+
+    getCommandlet(Node.class).install();
+    return super.doInstall(silent);
+  }
+
+  @Override
+  public void postInstall() {
```

Comment:
can you move this from `postInstall` to `postExtract` as discussed last Tuesday.
It is not wrong here but it is not as robust at the result would be inconsistent if an error occurred in the middle of this postInstall method as you then would already have an installation to be reused by IDEasy that is however incomplete.